src/PantheonTerminalWindow.vala

```diff
@@ -676,14 +678,29 @@ namespace PantheonTerminal {
 
             PantheonTerminal.saved_state.tabs = {};
 
+            int focus = 0 <= PantheonTerminal.saved_state.focused_tab < tabs.length
+                            ? PantheonTerminal.saved_state.focused_tab : tabs.length - 1;
```

Comment:
This is rather unclear and may not compile with older versions of valac. At least add a comment to explain the effect.

Consider using `focus = PantheonTerminal.saved_state.focused_tab.clamp (0, tabs.length - 1)`. This is clearer and means that a negative setting focusses the first tab.  As this situation is likely to be a rare corner case the exact behaviour is not that important.